src/CacheManager.Core/Internal/DictionaryCacheHandle.cs

```diff
@@ -84,6 +85,17 @@ public override bool Exists(string key, string region)
             return _cache.ContainsKey(fullKey);
         }
 
+        /// <inheritdoc />
+        public override IEnumerable<string> Keys(string pattern, string region)
+        {
+            if (region == null)
+                return _cache.Keys;
+
+            var skip = region.Length + 1;
+
+            return _cache.Keys.Where(k => k.StartsWith(region)).Select(k => k.Substring(skip)).FilterBy(pattern);
```

Comment:
Rewriting the pattern matching to take an offset would make it a lot more complex (It looks like it would require writing custom versions of StartsWith, Contains, etc. which is doable, but more than I was thinking of doing unless it showed up in a profile).